src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Rob Frawley 2nd <rmf@src.run>
+ */
+class MemcachedAdapter extends AbstractAdapter
+{
+    private static $defaultClientServer = array(
+        'host' => '127.0.0.1',
+        'port' => 11211,
+        'weight' => 100,
+    );
+
+    private $client;
+
+    public static function isSupported()
+    {
+        return extension_loaded('memcached') && version_compare(phpversion('memcached'), '2.2.0', '>=');
+    }
+
+    public function __construct(\Memcached $client, $namespace = '', $defaultLifetime = 0)
+    {
+        parent::__construct($namespace, $defaultLifetime);
+        $this->client = $client;
+    }
+
+    /**
+     * Factory method to create adapter setup with configured \Memcache client instance.
+     *
+     * Valid DSN values include the following:
+     *  - memcached://localhost                  : Specifies only the host (defaults used for port and weight)
+     *  - memcached://example.com:1234           : Specifies host and port (defaults weight)
+     *  - memcached://example.com:1234?weight=50 : Specifies host, port, and weight (no defaults used)
+     *
+     * Options are expected to be passed as an associative array with indexes of the option type with corresponding
+     * values as the option assignment. Valid options include any client constants, as described in the PHP manual:
+     *  - http://php.net/manual/en/memcached.constants.php
+     *
+     * @param string|null $dsn
+     * @param array       $opts
+     * @param string|null $persistentId
+     *
+     * @return MemcachedAdapter
+     */
+    public static function create($dsn = null, array $opts = array(), $persistentId = null)
+    {
+        $adapter = new static(new \Memcached($persistentId));
+        $adapter->setup($dsn ? array($dsn) : array(), $opts);
+
+        return $adapter;
+    }
+
+    /**
+     * @param string[] $dsns
+     * @param mixed[]  $opts
+     *
+     * @return bool
```

Comment:
Returns `$this` now: https://github.com/symfony/symfony/pull/20863/commits/3947b38f7352a310e83e5794fc3bbff61e94e2f0#diff-e01499fab5ad7f5e36517d157bc13e2fR72